specification/security_model.adoc

```diff
@@ -1,9 +1,438 @@
 [[security_model]]
-== Security and Threat Model
+== Security Model
+
+The CoVE-IO securiy model is built on the following assumptions:
+
+- The host platform physical devices are owned by untrusted domain software
+  components (e.g. the host VMM or the hypervisor) that are not part of any
+  TVM TCB and are thus untrusted by TVMs.
+- Only the TVM owner can assess of a TDI trustworthiness. Based on that
+  assessment, it explicitly accepts or rejects a TDI into its TCB.
+- A TDI may access a TVM confidential memory through DMA only when all the
+  following conditions are met:
+  * The TVM owner has explicitly allowed the TDI to access its confidential
+    memory by accepting it.
+  * The TDI is exclusively assigned to the TVM, i.e. it must not be shared
+    other TVMs or any host software component.
+- A TVM may access a TDI trusted MMIO space only when all the following
+  conditions are met:
+  * The TVM owner has explicitly accepted the TDI.
+  * The TDI is exclusively assigned to the TVM, i.e. it must not be shared
+    other TVMs or any host software component.
+- Until a TDI is accepted by the TVM:
+  * The TDI is not allowed to DMA into the TVM confidential memory.
+  * Trusted MMIO access to the TDI is blocked by TSM.
+  * Untrusted MMIO access to the TDI may still be allowed by the VMM for
+    ordinary VMs (if and only if the TDI is in the TDISP unlocked state).
+
+By means of the CoVE-IO guest ABI, TVMs are required to explicitly accept
+TDIs into their TCBs. Accepting or rejecting a TDI is a TVM specific decision,
+based on TVM specific set of verification policies and criteria. A TVM accepting
+a TDI does not imply that other TVMs on the same host platform would accept it
+as well.
+
+Multi-function, CoVE-IO-compliant devices may expose multiple TDIs, assigned to
+different TVMs. It is the DSM responsibility to guarantee isolation between all
+assigned TDIs, on a per-TVM basis.
+
+Each TDI must be exclusively assigned to no more than one TVM. However, a single
+TVM can simultaneously be assigned several TDIs, irrespective of whether they
+originate from the same physical device or not.
+
+As a TVM TCB does not include PCIe switches and bridges, a selective PCIe IDE
+stream must be setup to guarantee end-to-end confidentiality and integrity
+protection between a TVM and a TDI. The host VMM is responsible for setting up a
```

Comment:
Be more precise on the host VMM responsibilities. It does not configure the whole thing, only stream id and device IDE config space.

Reply:
Clarified.